README.md

```diff
@@ -36,14 +36,18 @@ below.  This has only been tested on Linux, but I think it will work on Windows
    for your operating system.
 2. Install the required Python packages, e.g., by running `pip install -r
    requirements.txt`
-3. For the LLM, install [Llama.cpp](https://github.com/ggerganov/llama.cpp), and compile it for your CPU or GPU. Edit the LLAMA_SERVER_PATH parameter in glados.py to match your installation path.
+3. For the LLM, set up your own OpenAI-compatible server andinstall [Llama.cpp](https://github.com/ggerganov/llama.cpp), by cloning it into `../llama.cpp` (relative to this folder), and compile it for your CPU or GPU. Edit the LLAMA_SERVER_PATH parameter in glados.py to match your installation path, if not cloning to `../llama.cpp`.
```

Comment:
Instead of this I'd just add https://github.com/ggerganov/llama.cpp as a git submodule.

Same for libwhisper and espeak-ng

Reply:
This was what I was thinking about doing. Then add in a script per platform for compilation. 

Ollama solves some issues for LLM inference, but you still have to compile Whisper.
The other option might be to inspect the platform and download release binaries for llama.cpp and whisper.cpp

Any opinions?

Reply:
IMO any device that can run this project in a usable manner, should have absolutely no issue building the relatively small codebase of neither libwhisper nor espeak-ng.

On my work MBP (M1 Ultra, 32GB RAM), it took about the same time to check out the projects as it did to build them - especially if you limit the build sequence to the bits we need (e.g. we don't need to compile everything in libwhisper, just libwhisper.so, and similarly we don't need the espeak-ng-data target, just the library).

Reply:
Also I think building from source is cleaner than relying on release binaries - especially given the recent `xz` debacle.

Reply:
OK, I have added llama and whisper as submodules, I guess I will add espeak-ng too.

I still haven't figure out the segfault issue with espeak-ng.
At this stage, I will probably move to calling the executable with subprocess. for a line of text, that a jump in processing from about 250us to 10ms, which sucks, but its not a breaking problem.